### ISSUE TITLE
[Doc]: 

### ISSUE THREAD

**Issue by JeroenvO** [matplotlib/matplotlib#25818]
### Documentation Link

https://matplotlib.org/devdocs/gallery/images_contours_and_fields/image_annotated_heatmap.html

### Problem

The outside border pixels of the heatmap 'leak' outside the grid on the x and y-axis borders, see this image:
https://matplotlib.org/devdocs/_images/sphx_glr_image_annotated_heatmap_002_2_00x.png
![Untitled](https://user-images.githubusercontent.com/5551571/236439310-ada54b27-6a63-44ec-8a5c-a787260b8000.png)

### Suggested improvement

This can be fixed by setting the spine color to white instead of turning them off:

Instead of 
https://github.com/matplotlib/matplotlib/blob/main/galleries/examples/images_contours_and_fields/image_annotated_heatmap.py#L152

Do: 
  `ax.spines[:].set_color('w')`

**Comment by tacaswell** [comment]
It is due to pixel snapping, if you run the code interactively and then resize the window the glitch comes and goes.  We must be snapping the image one way and the lines the other....

**Comment by Shree7676** [comment]
Hi @JeroenvO , @tacaswell 

I just updated the required changes as mentioned and cross verified it is working as expected 
please find the below screenshot for reference 

Can you please assign to me so that I can push the changes
![MatPlt_fix](https://github.com/matplotlib/matplotlib/assets/105912861/2874953f-f3ad-488e-ac63-462633bc4ac3)

**Comment by Shree7676** [comment]
I am new to Open-Source World and this would be my first contribution if allowed 

Below is the only change what I have done 
Eager to push waiting for Green signal
![matPlt_Changes](https://github.com/matplotlib/matplotlib/assets/105912861/1173a07e-60a8-4443-a1b0-44775432d154)


@JeroenvO

**Comment by JeroenvO** [comment]
Hi, looks good! Although it might be better to fix the pixel snapping as mentioned, I don't know how to do this and will probably be a lot of effort. So I suggest to do it like this, thanks for your help!
Before merging, please update the comment line, it now still says we remove the spines which is not true anymore. Also maybe reference this issue in the comment text as reference (I'm not sure if that is common to do in the matplotlib project)

**Comment by timhoffm** [comment]
The heatmap now looks correct. https://matplotlib.org/devdocs/gallery/images_contours_and_fields/image_annotated_heatmap.html

We've recently merged a number of position calculation improvements, which will be available in 3.11. I assume they have fixed this.

**Comment by QuLogic** [comment]
Yes, this was fixed by 89bbd1c7a442bea71be76347db5b3162ae206ef3.
